prompt_composer.py

```diff
@@ -0,0 +1,74 @@
+from typing import List, Dict
+
+
+def format_slide_text(slide_data: Dict[str, str]) -> str:
+    """Formats the text extracted from a slide for inclusion in the prompt."""
+    formatted_text = ""
+
+    if slide_data.get('title'):
+        formatted_text += f"Title: {slide_data['title']}\n"
+
+    if slide_data.get('subtitle'):
+        formatted_text += f"Subtitle: {slide_data['subtitle']}\n"
+
+    if slide_data.get('body'):
+        formatted_text += f"Body:\n{slide_data['body']}\n"
+
+    for j, table in enumerate(slide_data.get('tables', [])):
+        formatted_text += f"Table {j + 1}:\n"
+        for cell_text in table:
+            formatted_text += f"{cell_text}\n"
+
+    for j, smartart_text in enumerate(slide_data.get('smartart', [])):
+        formatted_text += f"SmartArt {j + 1}:\n{smartart_text}\n"
+
+    return formatted_text
+
+
+def single_slide_prompt(slide_data: Dict[str, str], slide_number: int) -> str:
+    """
+    Composes a prompt for a single slide in the presentation, including detailed analysis points.
+
+    Args:
+        slide_data (dict): A dictionary representing a slide in the presentation.
+        slide_number (int): The number of the slide in the presentation.
+
+    Returns:
+        str: The composed prompt for the slide.
+    """
+    prompt = f"Slide {slide_number} Analysis:\n"
+    prompt += format_slide_text(slide_data)
+    prompt += "\n"
+    prompt += "Please analyze this slide and provide detailed insights on the following aspects:\n\n"
+    prompt += "1. Key Message: What is the main message or purpose of this slide?\n"
```

Comment:
Better to use """all the text """ instead of concatenate strings like this.
triple quotation marks are not only for documentation